update-dockerhub-repo.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -e
+
+# Json file containing the description of the repository
+INPUT_JSON_FILE=$1
+
+if [ ! -f ${INPUT_JSON_FILE} ]; then
+    echo "File '${INPUT_JSON_FILE}' is missing. Exiting now..."
+    exit
+fi
+
+# DOCKERHUB_CREDS_USR and DOCKERHUB_CREDS_PSW must be present as environment variables
+# DOCKERHUB_CREDS_USR will typically be 'docker'
+if [ -z "$DOCKERHUB_CREDS_USR" ] || [ -z "$DOCKERHUB_CREDS_PSW" ] ; then
+    echo "Variables missing. Seems that you are running from a console. Prompting user and passwd"
+    read  -p 'Username: ' DOCKERHUB_CREDS_USR
+    read -sp 'Password: ' DOCKERHUB_CREDS_PSW
+    echo
+fi
+
+# Read Json file with data about the repository
+DOCKERHUB_REPO=`cat ${INPUT_JSON_FILE}`
+DESCRIPTION_SHORT=$(echo ${DOCKERHUB_REPO} | jq -r '.repository.description')
+DESCRIPTION_FULL=$(echo ${DOCKERHUB_REPO} | jq -r '.repository.full_description')
+
+# Get JWT token
+echo
+echo "Getting token"
+TOKEN=$(curl -v -XPOST \
+    -H "Content-Type: application/json" \
+    -d '{"username": "'${DOCKERHUB_CREDS_USR}'", "password": "'${DOCKERHUB_CREDS_PSW}'"}' \
+    https://hub.docker.com/v2/users/login/ | jq -r '.token')
```

Comment:
Needs some error handling, at least checking that `TOKEN` is non-empty.

As an alternative you could also consider stealing [this linuxkit code](https://github.com/linuxkit/linuxkit/blob/master/scripts/push-manifest.sh#L23...L56) which extracts the token from `.docker/config` (or the creds store). Since Jenkins calls this from within a `docker.WithRegistry` that ought to be present and usable.